src/box/lua/config/applier/box_cfg.lua

```diff
@@ -428,7 +435,119 @@ local function apply(config)
     box.cfg(box_cfg)
 end
 
+local function names_apply_f(configdata)
+    local name = 'config_apply_names'
+    fiber.self():name(name)
+
+    box.ctl.wait_rw()
+
+    -- Somebody else already applied all names.
+    if table.equals(configdata._missing_names, {}) then
+        return
+    end
+
+    box.begin()
+    for name, uuid in pairs(configdata._missing_names) do
+        if name == configdata:names().replicaset_name then
+            box.space._schema:insert{'replicaset_name', name}
+        else
+            local tuple = box.space._cluster.index.uuid:select(uuid)
+            box.space._cluster:update(tuple[1][1], {{'=', 3, name}})
+        end
+    end
+    box.commit()
+end
+
+local function names_box_cfg(cfg)
+    local ok, err = pcall(function()
+        box.cfg(cfg)
+    end)
+
+    -- We don't consider failed box.cfg a critical error,
+    -- delete a warning anyway. box.info shows correct info.
+    if not ok then
+        log.warn('Failed to apply name in box.cfg: %s', err)
+    end
+end
+
+local function names_on_commit(config, name)
+    -- Delete name from _missing_names and drop the associated alert.
+    config._configdata._missing_names[name] = nil
+    config:_alert_drop(name)
+
+    local watch_key = 'box_apply_names'
+    local names = box.watch_once(watch_key) or {}
+    table.insert(names, name)
+    box.broadcast(watch_key, names)
+end
+
+local function post_apply(config)
+    local configdata = config._configdata
+    if table.equals(configdata._missing_names, {}) then
+        return
+    end
+
+    for name, uuid in pairs(configdata._missing_names) do
```

Comment:
We cannot make alerts during box.cfg, as they might not be deleted, if master sends names during recovery process

Reply:
Added comment